sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -226,6 +234,11 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
       }
      """
   }
+
+  override def sql: Option[String] = for {
+    valueSQL <- child.sql
+    listSQL <- sequenceOption(hset.toSeq.map(Literal(_).sql))
+  } yield s"($valueSQL IN (${listSQL.mkString(", ")}))"
```

Comment:
Maybe we do not need to support InSet right now? It is only created by the optimizer. Also, values in the set are in our internal type, I am not sure if using `toString` works.